src/api/c/sparse.cpp

```diff
@@ -0,0 +1,395 @@
+/*******************************************************
+ * Copyright (c) 2014, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#include <af/sparse.h>
+#include <af/array.h>
+#include <af/algorithm.h>
+#include <sparse_t.hpp>
+#include <sparse.hpp>
+#include <handle.hpp>
+#include <backend.hpp>
+#include <err_common.hpp>
+#include <arith.hpp>
+#include <lookup.hpp>
+
+using namespace detail;
+using af::dim4;
+
+af_sparse_array getSparseHandle(const af_sparse_t sparse)
+{
+    af_sparse_t *sparseHandle = new af_sparse_t;
+    *sparseHandle = sparse;
+    return (af_sparse_array)sparseHandle;
+}
+
+af_sparse_t getSparse(const af_sparse_array sparseHandle)
+{
+    return *(af_sparse_t *)sparseHandle;
+}
```

Comment:
The definitions are in the header. The implementation can be here. Assuming the header is included in multiple places.